src/renderers/shared/reconciler/__tests__/ReactCompositeComponent-test.js

```diff
@@ -424,6 +424,35 @@ describe('ReactCompositeComponent', function() {
     expect(instance2.state.value).toBe(1);
   });
 
+  it('should warn about `setState` in getChildContext', function() {
+    var container = document.createElement('div');
+
+    var renderPasses = 0;
+
+    var Component = React.createClass({
+      getInitialState: function() {
+        return {value: 0};
+      },
+      getChildContext: function() {
+        if (this.state.value === 0) {
+          this.setState({ value: 1 });
+        }
+      },
+      render: function() {
+        renderPasses++;
+        return <div />;
+      },
+    });
+    expect(console.error.calls.length).toBe(0);
+    var instance = ReactDOM.render(<Component />, container);
+    expect(renderPasses).toBe(2);
+    expect(instance.state.value).toBe(1);
+    expect(console.error.calls.length).toBe(1);
+    expect(console.error.argsForCall[0][0]).toBe(
+      'Warning: setState(...): Cannot call setState inside getChildContext()'
```

Comment:
Nitpicking: let’s add `()` for consistency? As in `Cannot call setState() inside getChildContext()`.